schoolmatch.py

```diff
@@ -45,146 +35,100 @@ def string_truncate(strings_to_truncate, words_to_remove, truncated_list_name):
             modifying_string = modifying_string.strip()
         truncated_list_name.append(modifying_string)  
     return truncated_list_name
-#Will take a look at stop words next
+
+
+def school_match(school_type, output_type, rs_df, ospi_df):
+    '''
+    Takes the type of school (public or independent) as a string, a string 
+    "matched" or "unmatched" that determines whether the output is the schools 
+    that matched or those that didn't, the processed dataframe from 
+    Rainier Scholars, and the processed dataframe from OPSI using only the 
+    code, name, and city columns in that order and looks for matches by 
+    comparing the truncated names.
+    '''
+    ospi_code = []
+    ospi_name = []
+    ospi_city = []
+    ospi_type = []
+
+    for rs_idx, school in enumerate(rs_df.TruncName):
+        getcity = rs_df.iloc[rs_idx,2]
+        ospi_df_bycity = ospi_df[ospi_df.City == getcity] 
+        
+        best_school_so_far = np.nan
+        
+        for ospi_idx, item in enumerate(ospi_df_bycity.TruncName):
+            if school == item:
+                best_school_so_far = 0
+                best_school_so_far = ospi_idx
+                
+        if pd.isnull(best_school_so_far):
+            ospi_code.append(np.nan)
+            ospi_name.append(np.nan)
+            ospi_city.append(np.nan)
+            
+        else:
+            ospi_code.append(ospi_df_bycity.iloc[best_school_so_far,0])
+            ospi_name.append(ospi_df_bycity.iloc[best_school_so_far,1])
+            ospi_city.append(ospi_df_bycity.iloc[best_school_so_far,2])
+            ospi_type.append(school_type)
+    
+    ospi_matched_df = pd.DataFrame({"OSPI School Code": ospi_code,
+                                   "OSPI School School Name": ospi_name,
+                                   "OSPI City": ospi_city}, index = range(0,len(ospi_code)), 
+        columns=["OSPI School Code", "OSPI School School Name", "OSPI City"])
+    
+    rs_df.reset_index(drop=True, inplace=True)
+    ospi_matched_df.reset_index(drop=True, inplace=True)
+    
+    all_df = pd.concat([rs_df, ospi_matched_df], axis=1)
+    all_unmatched_df = all_df[all_df["OSPI City"].isnull()]
+    all_matched_df = all_df[all_df["OSPI City"].notnull()]
+
+    if output_type == "matched":

```

Comment:
In this function, you are doing all the work to create both the matched and unmatched versions of the df. You are throwing one away and returning the other.  Then, outside this function, you are calling it twice, once for each output type.  
It's better to have this function always return both output types.  
You can return them both as a tuple, like this:
return (all_matched_df, all_unmatched_df)

Then, you call the function like this:
(all_pub_matched_df, all_pub_unmatched_df) = school_match("public", rs_pub_df, 
                                                          ospi_raw_pub_df[["SchoolCode", "SchoolName", "City", "TruncName"]])